lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1447,26 +1471,47 @@ synthesizeObservedSetterBody(AccessorDecl *Set, TargetImpl target,
     SetterBody.push_back(Call);
   };
 
+  // Walk the didSet to check if we have a reference to the oldValue
+  auto foundOldValueRef = [&](const AccessorDecl *decl) {
```

Comment:
Please use an ASTWalker here to properly visit all child expressions. You’re only visiting statement expressions and not expressions inside property initializers, if statements, etc.

Reply:
Also, please factor this walker out into a request so we can run it when deciding if the observer should have an oldValue parameter or not.

Reply:
I have added a request!